conan/tools/env/environment.py

```diff
@@ -18,37 +18,43 @@ def environment_wrap_command(env_filenames, env_folder, cmd, subsystem=None,
     if not env_filenames:
         return cmd
     filenames = [env_filenames] if not isinstance(env_filenames, list) else env_filenames
-    bats, shs, ps1s = [], [], []
+    bats, shs, ps1s, fishes = [], [], [], []
 
-    accept = accepted_extensions or ("ps1", "bat", "sh")
+    accept = accepted_extensions or ("ps1", "bat", "fish", "sh")
     # TODO: This implemantation is dirty, improve it
     for f in filenames:
         f = f if os.path.isabs(f) else os.path.join(env_folder, f)
         if f.lower().endswith(".sh"):
             if os.path.isfile(f) and "sh" in accept:
                 f = subsystem_path(subsystem, f)
                 shs.append(f)
+        elif f.lower().endswith(".fish"):
+            if os.path.isfile(f) and "fish" in accept:
+                fishes.append(f)
         elif f.lower().endswith(".bat"):
             if os.path.isfile(f) and "bat" in accept:
                 bats.append(f)
         elif f.lower().endswith(".ps1") and "ps1" in accept:
             if os.path.isfile(f):
                 ps1s.append(f)
         else:  # Simple name like "conanrunenv"
-            path_bat = "{}.bat".format(f)
-            path_sh = "{}.sh".format(f)
-            path_ps1 = "{}.ps1".format(f)
+            path_bat = f"{f}.bat"
+            path_sh = f"{f}.sh"
+            path_ps1 = f"{f}.ps1"
+            path_fish = f"{f}.fish"
             if os.path.isfile(path_bat) and "bat" in accept:
                 bats.append(path_bat)
             if os.path.isfile(path_ps1) and "ps1" in accept:
                 ps1s.append(path_ps1)
             if os.path.isfile(path_sh) and "sh" in accept:
                 path_sh = subsystem_path(subsystem, path_sh)
                 shs.append(path_sh)
+            if os.path.isfile(path_fish) and "fish" in accept:
+                fishes.append(path_fish)
 
-    if bool(bats) + bool(shs) + bool(ps1s) > 1:
+    if bool(bats) + bool(shs) + bool(ps1s) + bool(fishes) > 1:
```

Comment:
adding bool as a number looks weird here...
I suppose any `bool` has only two values, `true` (1) and `false` (0).
therefore, a condition like `bool_var > 1` has no sense, as both valid values `true`/`false` are never greater than one.
I don't think we define `plus` operator for boolean values, so this code doesn't look good.
even if it works as expected, I'd suggest to re-write it to avoid computing sums of boolean values.

Reply:
You're right that this looks a bit iffy if you're not used to it, but in actuality, in Python Bool is a subtype of Int, so operations such as addition are well-defined https://docs.python.org/3/library/stdtypes.html#numeric-types-int-float-complex

https://godbolt.org/z/PdYvYhfco

Reply:
But you're right that there might be a more expressive way for this, thanks for the suggestion!